crates/matrix-sdk/src/send_queue.rs

```diff
@@ -72,51 +77,49 @@ impl SendQueue {
         room_q
     }
 
-    /// Enable the send queue for the entire client, i.e. all rooms.
+    /// Enable or disable the send queue for the entire client, i.e. all rooms.
+    ///
+    /// If we're disabling the queue, and requests were being sent, they're not
+    /// aborted, and will continue until a status resolves (error responses
+    /// will keep the events in the buffer of events to send later). The
+    /// disablement will happen before the next event is sent.
     ///
     /// This may wake up background tasks and resume sending of events in the
     /// background.
-    pub fn enable(&self) {
-        if self.client.inner.send_queue_data.globally_enabled.set_if_not_eq(true).is_some() {
-            debug!("globally enabling send queue");
-            let rooms = self.client.inner.send_queue_data.rooms.read().unwrap();
-            // Wake up the rooms, in case events have been queued in the meanwhile.
-            for room in rooms.values() {
-                room.inner.notifier.notify_one();
-            }
-        }
-    }
+    pub fn set_enabled(&self, enabled: bool) {
+        debug!("globally {}abling send queue", if enabled { "en" } else { "dis" });
```

Comment:
Nitpicking: It creates a string allocation every time `debug!` is called. Changing for:

```rust
if enabled {
    debug!("globally enabling send queue");
} else {
    debug!("globally disabling send queue");
}
```

would create zero string allocation, and will use a static string. I think it's preferable :-].

Reply:
I don't think it allocates, there's no reason the strings in the conditionals' bodies can't be static too.

Reply:
Replaced with         debug!(?enabled, "setting global send queue enablement");